packages/controllers/src/plugins/PluginController.test.ts

```diff
@@ -285,4 +286,62 @@ describe('PluginController Controller', () => {
       sourceCode,
     });
   });
+
+  it('can not delete existing plugins when using runExistinPlugins with a hydrated state', async () => {
+    expect.assertions(1);
+
+    const mockExecutePlugin = jest.fn();
+
+    const firstPluginController = new PluginController({
+      terminateAllPlugins: jest.fn(),
+      terminatePlugin: jest.fn(),
+      executePlugin: mockExecutePlugin,
+      getRpcMessageHandler: jest.fn(),
+      removeAllPermissionsFor: jest.fn(),
+      getPermissions: jest.fn(),
+      hasPermission: jest.fn(),
+      requestPermissions: jest.fn(),
+      closeAllConnections: jest.fn(),
+      messenger: new ControllerMessenger<any, any>().getRestricted({
+        name: 'PluginController',
+      }),
+      state: {
+        pluginStates: {},
+        inlinePluginIsRunning: false,
+        plugins: {
+          foo: {
+            isRunning: true,
+            initialPermissions: {},
+            permissionName: 'fooperm',
+            version: '0.0.1',
+            sourceCode: 'console.log("foo")',
+            name: 'foo',
+          },
+        },
+      },
+    });
+    // persist the state somewhere
+    const persistedState = getPersistentState<PluginControllerState>(
+      firstPluginController.state,
+      firstPluginController.metadata,
+    );
+    // create a new controller
+    const secondPluginController = new PluginController({
+      terminateAllPlugins: jest.fn(),
+      terminatePlugin: jest.fn(),
+      executePlugin: mockExecutePlugin,
+      getRpcMessageHandler: jest.fn(),
+      removeAllPermissionsFor: jest.fn(),
+      getPermissions: jest.fn(),
+      hasPermission: jest.fn(),
+      requestPermissions: jest.fn(),
+      closeAllConnections: jest.fn(),
+      messenger: new ControllerMessenger<any, any>().getRestricted({
+        name: 'PluginController',
+      }),
+      state: persistedState as unknown as PluginControllerState,
+    });
+    await secondPluginController.runExistingPlugins();
```

Comment:
I think we should check the value of `isRunning` for the plugin `foo` immediately before and after this call.